protocol/x/clob/keeper/process_single_match.go

```diff
@@ -432,18 +436,47 @@ func (k Keeper) persistMatchedOrders(
 	// keeper.
 
 	if err := k.subaccountsKeeper.TransferInsuranceFundPayments(ctx, insuranceFundDelta, perpetualId); err != nil {
-		return takerUpdateResult, makerUpdateResult, err
+		return takerUpdateResult, makerUpdateResult, affiliateRevSharesQuoteQuantums, err
+	}
+	perpetual, err := k.perpetualsKeeper.GetPerpetual(ctx, perpetualId)
```

Comment:
(In a follow-up PR) `ProcessSingleMatch` is making duplicate calls of `GetPerpetual`, let's make this call once as pass down to functions (completing the TODO item above). We previously made similar changes to optimize performance (at `UpdateSubaccounts` level)

Reply:
Agreed. Will add a todo